src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -99,7 +99,7 @@ private void EnsureSwitchGoverningExpressionAndDiagnosticsBound()
 
                 object key;
                 var constantValue = label.SwitchCaseLabelConstant;
-                if ((object)constantValue != null && !constantValue.IsBad)
+                if (constantValue is object { IsBad: false })
```

Comment:
💭 Will this even compile?

Reply:
Covered already by https://github.com/dotnet/roslyn/pull/39738#discussion_r344381734